copy/copy.go

```diff
@@ -465,12 +459,24 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 		}
 	}
 
+	// Check if the updates or a type conversion meaningfully changed the list of images.
+	updatedManifestList, err := list.Serialize()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Error encoding updated manifest list (%q: %#v)", list.MIMEType(), list.Instances())
+	}
+	originalManifestList, err := originalList.Serialize()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Error encoding original manifest list for comparison (%q: %#v)", originalList.MIMEType(), originalList.Instances())
+	}
+	listIsModified := !bytes.Equal(updatedManifestList, originalManifestList)
```

Comment:
(Absolutely non-blocking: This variable could be inlined into its only use.)

Reply:
Changed it.